src/Fmt_ast.ml

```diff
@@ -176,15 +176,20 @@ let parens_or_begin_end c ~loc =
       let str = String.lstrip (Source.string_at c.source loc) in
       if String.is_prefix ~prefix:"begin" str then `Begin_end else `Parens
 
-let wrap_fits_breaks_exp_if ?(space = true) c ~parens ~loc k =
-  match parens_or_begin_end c ~loc with
-  | `Parens -> wrap_fits_breaks_if ~space c.conf parens "(" ")" k
-  | `Begin_end -> wrap_fits_breaks_exp_begin_end ~parens k
+let wrap_fits_breaks_exp_if ?(space = true) ?(disambiguate = false) c
+    ~parens ~loc k =
+  if Poly.(parens_or_begin_end c ~loc = `Begin_end) then
+    wrap_fits_breaks_exp_begin_end ~parens k
+  else if disambiguate && c.conf.disambiguate_non_breaking_match then
```

Comment:
To make this a little more readable you could revert to using a match and write the disambiguate case like this: `` `Parens when disambiguate && c.conf.disambiguate_non_breaking_match``.